language/go/resolve.go

```diff
@@ -288,11 +291,23 @@ func resolveExternal(moduleMode bool, rc *repo.RemoteCache, imp string) (label.L
 		pkg = pathtools.TrimPrefix(impWithoutSemver, prefix)
 	}
 
-	return label.New(repo, pkg, defaultLibName), nil
+	// Determine what naming convention is used by the go_repository rule.
+	// If none is specified with "build_naming_convention", the either naming convention
+	// can be used, so we'll default to whatever's used in the current workspace to avoid churn.
+	nc := gc.goNamingConvention
+	if gc.repoNamingConvention != nil {
```

Comment:
Nit: I think `gc.repoNamingConvention` will always be non-nil at this point, so this condition can be removed.

`Configure` should be called in the root directory before any call to `Resolve`.